counter/src/main/java/org/infinispan/counter/impl/strong/StrongCounterKey.java

```diff
@@ -0,0 +1,94 @@
+package org.infinispan.counter.impl.strong;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.counter.api.StrongCounter;
+import org.infinispan.counter.impl.entries.CounterKey;
+import org.infinispan.counter.impl.externalizers.ExternalizerIds;
+import org.infinispan.util.ByteString;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * The key to store in the {@link org.infinispan.Cache} used by {@link StrongCounter}.
+ *
+ * @author Pedro Ruivo
+ * @see StrongCounter
+ * @since 9.0
+ */
+public class StrongCounterKey implements CounterKey {
```

Comment:
hmmm... I don't like. It would complicate it a lot because of the marshalling, I would have to expose its internals and I would have to duplicate code.